src/connectivity/mod.rs

```diff
@@ -968,7 +968,7 @@ impl<'py> FromPyObject<'py> for TargetNodes {
         if let Ok(int) = ob.extract::<usize>() {
             Ok(Self::Single(NodeIndex::new(int)))
         } else {
-            let mut target_set: HashSet<NodeIndex> = HashSet::new();
```

Comment:
revert this change as we already use https://docs.rs/hashbrown/latest/hashbrown/hash_set/struct.HashSet.html

`hashbrown` already uses foldhash by defuault AFAIK. and it can change with time, like with `ahash` -> `foldhash`.